mira/modeling/amr/stockflow.py

```diff
@@ -0,0 +1,193 @@
+            else:
+                param_dict['distribution'] = {
+                    'type': param.distribution.type,
+                    'parameters': param.distribution.parameters,
+                }
+            if param.concept and param.concept.units:
+                param_dict['units'] = {
+                    'expression': str(param.concept.units.expression),
+                    'expression_mathml': expression_to_mathml(
+                        param.concept.units.expression.args[0]),
+                }
+            self.parameters.append(param_dict)
+
+        link_id = 1
+        for idx, flow in enumerate(model.transitions.values()):
+            fid = flow.template.name \
+                if flow.template.name else f"t{idx + 1}"
+            # fixme: get grounding for transition
+            flow_dict = {"id": fid}
+            flow_dict['name'] = flow.template.display_name
+            flow_dict['upstream_stock'] = flow.consumed[0].concept.name
+            flow_dict['downstream_stock'] = flow.produced[0].concept.name
+
+            if flow.template.rate_law:
+                rate_law = flow.template.rate_law.args[0]
+                formatted_rate_law = format_rate_law(model, rate_law)
+                flow_dict['rate_expression'] = str(formatted_rate_law)
+                flow_dict['rate_expression_mathml'] = expression_to_mathml(formatted_rate_law)
+
+            self.flows.append(flow_dict)
+
+            for symbol in flow.template.rate_law.free_symbols:
+                link_dict = {'id': f'link{link_id}'}
+                str_symbol = str(symbol)
+                if str_symbol in model.parameters:
+                    str_symbol = str_symbol[2:]
+
+                link_dict['source'] = str_symbol
+                link_dict['target'] = fid
+                link_id += 1
+                self.links.append(link_dict)
+
+    def to_json(self):
+        return {
+            'header': {
+                'name': self.model_name,
+                'schema': '',
+                'description': self.model_description,
+                'schema_name': 'stockflow',
+                'model_version': '0.1',
+            },
+            'properties': self.properties,
+            'model': {
+                'flows': self.flows,
+                'stocks': self.stocks,
+                'auxiliaries': self.auxiliaries,
+                'links': self.links
+            },
+            'semantics': {'ode': {
+                'parameters': self.parameters,
+                'initials': self.initials,
+                'observables': self.observables,
+                'time': self.time if self.time else {'id': 't'}
+            }},
+            'metadata': self.metadata,
+        }
+
+
+def format_rate_law(model, rate_law) -> sympy.Expr:
+    local_dict = {}
+    for symbol in rate_law.free_symbols:
+        str_symbol = str(symbol)
+        if str_symbol in model.parameters:
+            local_dict[str_symbol] = sympy.Symbol(str_symbol[2:])
```

Comment:
This would have to be updated to use the actual generated auxiliary names rather then just `str_symbol[2:]`.